docs/persistence.rst

```diff
@@ -106,6 +106,365 @@ database record will also have its ID changed. Here is example::
+
+    $m->strict_types = true;
+    $m->load_normalization = false;
+
+Setting `strict_types` to false, will still disable any type-casting
+and store exact values you specify regardless of type. If you switch
+on `load_normalization` then the values will also be normalized
+as they are loaded from the database. Normally you should only
+do that if you're storing values into database by other means and
+not through Agile Data.
+
+Final field flag that is worth mentioning is called `read_only`
+and if set, then value of a field may not be modified directly::
+
+    $m->addField('ref_no', ['read_only' => true]);
+    $m->load(123);
+
+    $m['ref_no']; // perfect for reading field that is populated by trigger.
+
+    $m['ref_no'] = 'foo'; // exception
+
+Note that `read_only` can still have a default value::
+
+    $m->addField('created', [
+        'read_only' => true,
+        'type'      => 'datetime',
+        'default'   => new DateTime()
+    ]);
+
+    $m->save();  // stores creation time just fine and also will loade it.
+
+
+.. note:: If you hae been following our "Domain" vs "Persistence" then
+    you can probably see that all of the above functionality described
+    in this section apply only to the "Domain" model.
+
+Typecasting
+-----------
+
+Typecasting is envoked when you are attempting to save or load the
+record. Unlike strict types and normalization, typecasting is a
+persistence-specific operation.
+
+Typecasting is necessary to save the values inside the database and
+restore them back just as they were before. When modifying a
+record, typecasting will only be invoked on the fields which were
+dirty.
+
+Join construct will single out the fields that needs to be stored
+in a separate table and will typecast them before persisting.
+
+You must remember that type-castin is a two-way operation. If
+you introducing your own types, you will need to make sure they
+can be saved and loaded correctly. 
+
+There are also a lot of flexiblity on how exactly you typecast
+your values to match your database requirements.
+
+
+For 
+SQL databases it's essential to convert date formats, booleans
+and pack arrays::
+
+    $m->addField('registered', ['type'=>'date']);
+
+    $m['registered'] = time();   // will cast to DateTime
+
+    $m->save();   // will convert into YYYY-MM-DD
+
+Some formats such as `date`, `time` and `datetime` may have
+additional options to it::
+
+    $m->addField('registered', [
+        'type'=>'date',
+        'datetime_class'=>'Carbon\Carbon',
+        'persist_format'=>'d/m/Y',
+    ]);
+
+    $m['registered'] = time();   // will cast to Carbon
+
+    $m->save();   // will convert into DD/MM/YYYY in SQL
+
+Here is another example with booleans::
+
+    $m->addField('is_married', [
+        'type' => boolean',
+        'enum' => ['Yes', 'No']
+    ]);
+
+    $m['is_married'] = 'Yes';  // type-casts into true
+    $m['is_married'] = true;   // better way!
+
+    $m->save();   // stores as "Yes".
```

Comment:
Will switch Yes/No in docs, source and tests because [0=>'Yes', 1=>'No'] is misleading.
It should better be 0=false=No and 1=true=Yes.